### DISCUSSION TITLE
Identify reports a PDF as AI

### DISCUSSION THREAD

**Discussion by martinkoch-geniebelt** [ImageMagick/ImageMagick#6724] [Contribute]
### Description
 `identify` run on a PDF file generated by Adobe Illustrator reports that the type is AI, rather than PDF which is what I would expect. 

### Steps to Reproduce
On my MacOS Catalina 10.15.7, ImageMagick 7.0.10-56 Q16 x86_64, `identify 2pages.pdf` yields

```
$ identify 2pages.pdf
2pages.pdf[0] AI 576x288 576x288+0+0 16-bit sRGB 2593B 0.000u 0:00.001
2pages.pdf[1] AI 576x288 576x288+0+0 16-bit sRGB 2602B 0.000u 0:00.001
```

[2pages.pdf](https://github.com/ImageMagick/ImageMagick/files/5780894/2pages.pdf)

(Note: I tried on 
* mac / docker / ubuntu 20.04 / ImageMagick 6.9.10-23 Q16 x86_64 and got the expected result. 
```
2pages.pdf[0] PDF 560x288 560x288+0+0 16-bit sRGB 8050B 0.000u 0:00.000
2pages.pdf[1] PDF 560x288 560x288+0+0 16-bit sRGB 7067B 0.000u 0:00.000
```
* mac / docker / debian 10 / ImageMagick 7.0.10-55 Q16 x86_64 and got the incorrect result:
```
2pages.pdf[0] AI 576x288 576x288+0+0 16-bit sRGB 2591B 0.000u 0:00.001
2pages.pdf[1] AI 576x288 576x288+0+0 16-bit sRGB 2600B 0.000u 0:00.001
```
)

**Comment by dlemstra** [comment]
This is happening due to this check: https://github.com/ImageMagick/ImageMagick/blob/8d8a01057663b11c872589ebc3038f93236fe44a/coders/pdf.c#L731

**Comment by seanthingee** [comment]
I just stumbled onto this too... is this not a bug then? My software relies on `format` response to detect the file type for further processing or rejection. 

_Edit with more info_:

The problematic PDF I'm dealing with shows this in the verbose output:

```
Format: AI (Adobe Illustrator CS2)
Mime type: application/pdf
```

The _non_-problematic PDF shows this: 
```
Format: PDF (Portable Document Format)
Mime type: application/pdf
```

**Comment by Berschmi** [comment]
I ran into the same problem. 

ImageMagick 7.0.3-8 Q16 x64 works correctly:
`Format: PDF (Portable Document Format)`
`Mime type: application/pdf`

ImageMagick 7.1.1-27 Q16-HDRI x64 reports 'AI' as format:
`Format: PDF (Portable Document Format)`
`Format: AI (Adobe Illustrator CS2)`

Please give feedback why this is issue was converted into a discussion and is not qualified as a bug.
If needed, I can provide a PDF that shows the issue.

**Comment by dlemstra** [comment]
We don't consider this a bug.

**Reply by seanthingee** [reply]
> We don't consider this a bug.

Please elaborate on your decision.

**Reply by dlemstra** [reply]
Because this is the method that we use to detect if a file is a `.ai` file and that seems to work with the test files that I have.
